### PR TITLE
Use package manager

### PR DESCRIPTION
**Pull request by robkooper**
Do not set repos = ... in install.packages calls. We should not be overwriting the user- or system-specific setting in .Rprofile. In particular, this can produce insidious bugs when interacting with Docker containers that are configured to use specific CRAN snapshots. We should defer to the Docker R image configuration to handle this, unless strictly necessary.
Undo manual installing rlang and testthat in GH actions . If we defer to the correct Docker images with the correct fixed RStudio package manager snapshots, this should not be necessary

This requires a modified version of pecan/depends:4.0.2 (which i pushed) to pass the build.

This also removes all travis pieces.

## Review Time Estimate

- [X] Immediately
- [ ] Within one week
- [ ] When possible

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [X] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by mdietze — APPROVED** [review]
This looks good to me but would love for @ashiklom or @infotroph to comment

**Comment by robkooper** [comment]
This should address #2769

File: .travis.yml [hunk 1]
```diff
@@ -1,88 +0,0 @@
-language: r
```
Comment (infotroph) [inline .travis.yml:1]: All this code will be lost in time like tears in rain...

...and good riddance ;)

File: book_source/02_demos_tutorials_workflows/05_developer_workflows/04-testing.Rmd [hunk 2]
```diff
@@ -124,49 +124,26 @@ The `batch_run.R` script can take the following command-line arguments:
...
-* Runs package unit tests (the same ones you run locally with `make test` or `devtools::test(pkgname)`).
-    - As discussed in [Unit testing](#developer-testing-unit), these tests should run quickly and test individual components in relative isolation.
-    - Any test that calls the `skip_on_ci` function will be skipped. This is useful for tests that need to run for a very long time (e.g. large data product downloads) or require resources that aren't available on Travis (e.g. specific models), but be sure to run these tests locally before pushing your code!
-* Runs R package checks (the same ones you run locally with `make check` or `devtools::check(pkgname)`), skipping tests and documentation rebuild because we just did those in the previous steps.
-    - Any ERROR in the check output will stop the build immediately.
-    - If there are no ERRORs, any WARNINGs or NOTEs are compared against a stored historic check result in `<package>/tests/Rcheck_reference.log`. If the package has no stored reference result, all WARNINGs and NOTEs are considered newly added and reported as build failures.
```
Comment (infotroph) [inline book_source/02_demos_tutorials_workflows/05_developer_workflows/04-testing.Rmd:147]: I believe this section is the only written documentation of how the cached check comparisons work. Should it be retained (possibly in another location)?

**Review by infotroph — CHANGES_REQUESTED** [review]

File: book_source/02_demos_tutorials_workflows/05_developer_workflows/04-testing.Rmd (hunk 2 above)
Comment (robkooper) [reply book_source/02_demos_tutorials_workflows/05_developer_workflows/04-testing.Rmd:147]: use to big of an axe.

**Comment by robkooper** [comment]
4.0.3 uses latest so it will no have a consistent build, i feel that it is ok if that breaks and note make it a required pass.

File: book_source/02_demos_tutorials_workflows/05_developer_workflows/04-testing.Rmd (hunk 2 above)
Comment (robkooper) [reply book_source/02_demos_tutorials_workflows/05_developer_workflows/04-testing.Rmd:147]: added text back

**Review by infotroph — APPROVED** [review]

**Review by ashiklom — APPROVED** [review]

File: apps/api/Dockerfile [hunk 3]
```diff
@@ -18,9 +18,9 @@ EXPOSE 8000
 RUN   apt-get update \
   &&  apt-get install libsodium-dev -y \
   &&  rm -rf /var/lib/apt/lists/* \
-  &&  Rscript -e "devtools::install_version('promises', '1.1.0', repos = 'http://cran.rstudio.com')" \
-  &&  Rscript -e "devtools::install_version('webutils', '1.1', repos = 'http://cran.rstudio.com')" \
-  &&  Rscript -e "install.packages('pool', repos = 'http://cran.rstudio.com')" \
+  &&  Rscript -e "devtools::install_version('promises', '1.1.0')" \
+  &&  Rscript -e "devtools::install_version('webutils', '1.1')" \
```
Comment (ashiklom) [inline apps/api/Dockerfile:22]:
```suggestion
  &&  Rscript -e "devtools::install_version('promises', '1.1.0')" \
  &&  Rscript -e "devtools::install_version('webutils', '1.1')" \
```

File: apps/api/Dockerfile [hunk 4]
```diff
@@ -18,9 +18,9 @@ EXPOSE 8000
...
+  &&  Rscript -e "install.packages('pool')" \
   &&  Rscript -e "devtools::install_github('rstudio/swagger')" \
   &&  Rscript -e "devtools::install_github('rstudio/plumber')"
```
Comment (ashiklom) [inline apps/api/Dockerfile:25]: Ditto here.

```suggestion
  &&  Rscript -e "remotes::install_github('rstudio/swagger')" \
  &&  Rscript -e "remotes::install_github('rstudio/plumber')"
```

File: docker/depends/Dockerfile [hunk 5]
```diff
@@ -39,8 +39,8 @@ RUN apt-get update \
 # INSTALL DEPENDENCIES
 # ----------------------------------------------------------------------
 COPY pecan.depends.R /
-RUN  Rscript -e "install.packages(c('devtools'), repos = 'http://cran.rstudio.com')" \
-  && Rscript -e "devtools::install_version('roxygen2', '7.0.2', repos = 'http://cran.rstudio.com')" \
+RUN  Rscript -e "install.packages(c('devtools'))" \
+  && Rscript -e "devtools::install_version('roxygen2', '7.0.2')" \
```
Comment (ashiklom) [inline docker/depends/Dockerfile:43]: Ditto here:

```suggestion
RUN  Rscript -e "install.packages(c('remotes'))" \
  && Rscript -e "remotes::install_version('roxygen2', '7.0.2')" \
```

**Comment by ashiklom** [comment]
Looks good, thanks! Not gonna hold up the PR for this, but for future reference, I'd probably recommend switching to `remotes::install_*` (all my suggested edits) -- that's basically (exactly?) what `devtools::install_*` calls under the hood. But, the benefit of `remotes` is that it doesn't pull in (i.e., tries to update) all the other `devtools` packages, which, AFAICT, was the cause of all these issues to begin with (we were installing a more recent version of `devtools` but with an older version of `testthat` -- that would have been avoided if we'd just used `remotes` instead, which has no connection to `testthat`).

**Comment by robkooper** [comment]
Merging this, will create a new PR to do a the conversion from devtools::install to remotes::install

**Pull request merged**